In [1]:
import pandas as pd

In [2]:
data = 'Data_Original.csv'
df = pd.read_csv(data)

In [3]:
df

id nombre  edad  altura     ciudad color_ojos fecha_cumpleaños
0    986  Pedro  36.0     157  Barcelona       Azul          06/2004
1    613  Luisa  63.0     158    Sevilla     Marrón          12/1985
2    263   Sara   NaN     165   Zaragoza      Negro          05/2004
3    452   Sara   NaN     182   Zaragoza       Azul          08/2004
4    516    Ana  70.0     144     Madrid      Verde          11/1992
..   ...    ...   ...     ...        ...        ...              ...
995  106   Juan   NaN     153   Valencia       Azul          03/1981
996  616   Juan  50.0     161  Barcelona     Marrón          01/2003
997  965  Elena   NaN     174   Zaragoza      Verde          08/2005
998  483  Pablo  65.0     147   Zaragoza        NaN          04/2003
999  896  Pablo  56.0     187    Sevilla       Azul          05/1992

[1000 rows x 7 columns]

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   id                1000 non-null   int64  
 1   nombre            900 non-null    object 
 2   edad              901 non-null    float64
 3   altura            1000 non-null   int64  
 4   ciudad            899 non-null    object 
 5   color_ojos        900 non-null    object 
 6   fecha_cumpleaños  904 non-null    object 
dtypes: float64(1), int64(2), object(4)
memory usage: 54.8+ KB


*La función ***df.info()*** en Pandas proporciona un resumen conciso del DataFrame llamado df. Indica que tiene 1000 filas (de índice 0 a 999) y 7 columnas: id (entero, sin valores nulos), nombre (objeto/string, 100 valores nulos), edad (float, 99 valores nulos), altura (entero, sin valores nulos), ciudad (objeto/string, 101 valores nulos), color_ojos (objeto/string, 100 valores nulos) y fecha_cumpleaños (objeto/string, 96 valores nulos). Además, informa sobre los tipos de datos presentes (1 float64, 2 int64, 4 object) y el uso de memoria del DataFrame (aproximadamente 54.8 KB).*

In [5]:
df.describe()

id        edad       altura
count  1000.00000  901.000000  1000.000000
mean    500.30800   44.788013   170.484000
std     287.93665   16.884790    20.060612
min       1.00000    5.000000   100.000000
25%     250.75000   30.000000   153.000000
50%     500.50000   45.000000   169.000000
75%     748.25000   58.000000   186.000000
max    1000.00000  120.000000   300.000000

*La función ***df.describe()*** genera estadísticas descriptivas de las columnas numéricas del DataFrame df. Para las columnas id, edad y altura, muestra la cantidad de valores no nulos (count), la media, la desviación estándar (std), el valor mínimo (min), los cuartiles (25%, 50% o mediana, 75%) y el valor máximo (max). Esto revela información sobre la distribución de los datos numéricos, como los rangos, las tendencias centrales y la dispersión. Por ejemplo, la edad promedio es de aproximadamente 44.79 años, con un rango de 5 a 120 años, mientras que la altura promedio es de 170.48 con un rango de 100 a 300. La diferencia en el conteo para la columna 'edad' (901) en comparación con 'id' y 'altura' (1000) confirma la presencia de valores faltantes en esa columna.*

In [6]:
nulos = df.isnull().sum()
print("Valores nulos por columna:")
print(nulos)

Valores nulos por columna:
id                    0
nombre              100
edad                 99
altura                0
ciudad              101
color_ojos          100
fecha_cumpleaños     96
dtype: int64


*Este código calcula y muestra la cantidad de valores nulos presentes en cada columna del DataFrame df. Primero, df.isnull() crea un DataFrame booleano indicando qué valores son nulos. Luego, .sum() suma los True (interpretados como 1) por cada columna, resultando en una Serie llamada nulos que contiene el conteo de valores nulos por columna. Finalmente, se observa un encabezado descriptivo seguido de la Serie nulos, mostrando que columnas como nombre, edad, ciudad, color_ojos y fecha_cumpleaños tienen valores faltantes (100, 99, 101, 100 y 96 respectivamente), mientras que id y altura no tienen ningún valor nulo.*

In [7]:
# Eliminar valores nulos
df_sin_nulos = df.dropna()
df_sin_nulos.isnull().sum()

id                  0
nombre              0
edad                0
altura              0
ciudad              0
color_ojos          0
fecha_cumpleaños    0
dtype: int64

*Este código busca eliminar las filas que contienen al menos un valor nulo del DataFrame original df. Primero, df.dropna() crea un nuevo DataFrame llamado df_sin_nulos que contiene solo las filas del DataFrame original donde no hay ningún valor nulo en ninguna de las columnas. Luego, df_sin_nulos.isnull().sum() se aplica a este nuevo DataFrame para contar la cantidad de valores nulos en cada columna. La salida muestra que para todas las columnas (id, nombre, edad, altura, ciudad, color_ojos, fecha_cumpleaños) el número de valores nulos es 0, lo que confirma que la función dropna() eliminó exitosamente todas las filas que contenían al menos un valor faltante.*

In [ ]:
print("Número de filas antes de la eliminación:", len(df)) 
print("Número de filas después de la eliminación:", len(df_sin_nulos))

Número de filas antes de la eliminación: 1000
Número de filas después de la eliminación: 589


*Este código muestra el impacto de la operación dropna(), revelando que de las 1000 filas originales, solo 589 filas no contenían ningún valor nulo en ninguna de sus columnas. Esto implica que 1000 - 589 = 411 filas fueron eliminadas porque contenían al menos un valor faltante.*

In [9]:
# Renombrar las variables

df = df_sin_nulos.rename(columns={
    'id': 'Usuario',
    'nombre': 'Nombre',
    'edad': 'Edad',
    'altura':'Altura',
    'ciudad': 'Ciudad',
    'color_ojos': 'Ojos',
    'fecha_cumpleaños': 'Cumpleaños',})

df

Usuario  Nombre  Edad  Altura     Ciudad    Ojos Cumpleaños
0        986   Pedro  36.0     157  Barcelona    Azul    06/2004
1        613   Luisa  63.0     158    Sevilla  Marrón    12/1985
4        516     Ana  70.0     144     Madrid   Verde    11/1992
5        208   María  25.0     181   Valencia  Marrón    04/1994
6        833    Juan  65.0     191  Barcelona  Marrón    05/1981
..       ...     ...   ...     ...        ...     ...        ...
990       65   Pablo  21.0     175   Zaragoza    Azul    04/1987
992      930   Luisa  19.0     140    Sevilla   Negro    02/2005
994      954  Carlos  42.0     155  Barcelona   Negro    11/2003
996      616    Juan  50.0     161  Barcelona  Marrón    01/2003
999      896   Pablo  56.0     187    Sevilla    Azul    05/1992

[589 rows x 7 columns]

*Este código utiliza el método .rename() en el DataFrame df_sin_nulos para cambiar los nombres de sus columnas. Se proporciona un diccionario donde las claves son los nombres originales de las columnas ('id', 'nombre', 'edad', 'altura', 'ciudad', 'color_ojos', 'fecha_cumpleaños') y los valores son los nuevos nombres deseados ('Usuario', 'Nombre', 'Edad', 'Altura', 'Ciudad', 'Ojos', 'Cumpleaños'). El DataFrame resultante con las columnas renombradas se asigna de nuevo a la variable df. La salida muestra una vista parcial del DataFrame con los nuevos nombres de columna y confirma que mantiene las 589 filas y 7 columnas resultantes de la eliminación de valores nulos.*

In [10]:
dtype_original = df.dtypes
print(df.dtypes)

Usuario         int64
Nombre         object
Edad          float64
Altura          int64
Ciudad         object
Ojos           object
Cumpleaños     object
dtype: object


*Este código extrae los tipos de datos de cada columna del DataFrame df utilizando el atributo .dtypes y los almacena en la variable dtype_original. Posteriormente, imprime esta Serie que muestra el nombre de cada columna junto con su respectivo tipo de dato. La salida revela que la columna 'Usuario' y 'Altura' son de tipo entero (int64), 'Edad' es de tipo float (float64), mientras que 'Nombre', 'Ciudad', 'Ojos' y 'Cumpleaños' son de tipo 'object', lo que generalmente indica que contienen cadenas de texto o una mezcla de tipos.*

In [11]:
df_numerico = df_sin_nulos.select_dtypes(include=['float64', 'int64'])

Q1 = df_numerico.quantile(0.25)
Q3 = df_numerico.quantile(0.75)
IQR = Q3 - Q1
df_numerico = df_numerico[~((df_numerico < (Q1 - 1.5 * IQR)) | (df_numerico > (Q3 + 1.5 * IQR))).any(axis=1)]

print(f"Filas antes de eliminar valores atípicos: {df.shape[0]}")
print(f"Filas después de eliminar valores atípicos: {df_numerico.shape[0]}")
print("DataFrame después de eliminar valores atípicos:")
print(df_numerico)

Filas antes de eliminar valores atípicos: 589
Filas después de eliminar valores atípicos: 582
DataFrame después de eliminar valores atípicos:
      id  edad  altura
0    986  36.0     157
1    613  63.0     158
4    516  70.0     144
5    208  25.0     181
6    833  65.0     191
..   ...   ...     ...
990   65  21.0     175
992  930  19.0     140
994  954  42.0     155
996  616  50.0     161
999  896  56.0     187

[582 rows x 3 columns]


*Este código identifica y elimina valores atípicos de las columnas numéricas ('id', 'edad', 'altura') del DataFrame df_sin_nulos. Primero, selecciona solo las columnas numéricas en un nuevo DataFrame df_numerico. Luego, calcula el primer cuartil (Q1), el tercer cuartil (Q3) y el Rango Intercuartílico (IQR) para cada columna numérica. Utiliza estos valores para definir los límites de los valores atípicos (fuera de [Q1 - 1.5 \times IQR, Q3 + 1.5 \times IQR]) y filtra el DataFrame df_numerico para mantener solo las filas donde ningún valor en las columnas numéricas se considera un atípico. Finalmente, imprime el número de filas antes y después de la eliminación de los atípicos, mostrando que el DataFrame pasó de 589 a 582 filas, y presenta una muestra del DataFrame resultante sin los valores atípicos detectados.*

In [12]:
df_antes = df.copy()

df.drop_duplicates(inplace=True)

print(f"Número de filas con duplicados: {df_antes.shape[0]}")
print(f"Número de filas sin duplicados: {df.shape[0]}")

# Comprobar si hay duplicados restantes
duplicated = df.duplicated().sum()
print(f"Duplicados restantes: {duplicated}")

Número de filas con duplicados: 589
Número de filas sin duplicados: 581
Duplicados restantes: 0


*Este código comienza creando una copia del DataFrame df en df_antes. Luego, utiliza el método drop_duplicates(inplace=True) para eliminar las filas duplicadas directamente del DataFrame df. A continuación, imprime el número de filas que tenía el DataFrame antes de la eliminación (589) y el número de filas que quedaron después de eliminar las duplicadas (581). Finalmente, comprueba si quedan duplicados en el DataFrame resultante utilizando df.duplicated().sum(), imprimiendo que el número de duplicados restantes es 0, lo que confirma que la eliminación fue exitosa y se removieron 8 filas duplicadas.*